From jacmet@sunsite.dk  Mon Mar 15 16:37:46 2010
From: Peter Korsgaard <jacmet@sunsite.dk>
Date: Fri, 12 Mar 2010 15:55:28 +0100
Subject: USB: gadget: f_mass_storage::fsg_bind(): fix error handling
To: m.nazarewicz@samsung.com, gregkh@suse.de, linux-usb@vger.kernel.org
Cc: Peter Korsgaard <jacmet@sunsite.dk>
Message-ID: <1268405728-6547-1-git-send-email-jacmet@sunsite.dk>


Contrary to the comment in fsg_add, fsg_bind calls fsg_unbind on errors,
which decreases refcount and frees the fsg_dev structure, causing trouble
when fsg_add does the same.

Fix it by simply leaving up cleanup to fsg_add().

Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
Acked-by: Michal Nazarewicz <m.nazarewicz@samsung.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/f_mass_storage.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/usb/gadget/f_mass_storage.c
+++ b/drivers/usb/gadget/f_mass_storage.c
@@ -2954,7 +2954,6 @@ static int __init fsg_bind(struct usb_co
 autoconf_fail:
 	ERROR(fsg, "unable to autoconfigure all endpoints\n");
 	rc = -ENOTSUPP;
-	fsg_unbind(c, f);
 	return rc;
 }
 
